vectordraw/public/js/studio_edit.js

```diff
@@ -0,0 +1,139 @@
+            });
+        }
+
+    });
+
+    var studio_submit = function(data) {
+        var handlerUrl = runtime.handlerUrl(element, 'submit_studio_edits');
+        runtime.notify('save', {state: 'start', message: gettext("Saving")});
+        $.ajax({
+            type: "POST",
+            url: handlerUrl,
+            data: JSON.stringify(data),
+            dataType: "json",
+            global: false  // Disable Studio's error handling that conflicts with studio's notify('save') and notify('cancel') :-/
+        }).done(function(response) {
+            runtime.notify('save', {state: 'end'});
+        }).fail(function(jqXHR) {
+            if (jqXHR.responseText) { // Is there a more specific error message we can show?
+                try {
+                    errorMessage = JSON.parse(jqXHR.responseText).error;
+                    if (typeof errorMessage === "object" && errorMessage.messages) {
+                        // e.g. {"error": {"messages": [{"text": "Unknown user 'bob'!", "type": "error"}, ...]}} etc.
+                        errorMessage = $.map(errorMessage.messages, function(msg) { return msg.text; }).join(", ");
+                    }
+                } catch (error) { errorMessage = jqXHR.responseText.substr(0, 300); }
+            }
+            runtime.notify('error', {title: gettext("Unable to update settings"), message: errorMessage});
+        });
+    };
+
+    return {
+
+        getContents: function(fieldName) {
+            return _.findWhere(fields, {name: fieldName}).val();
+        },
+
+        save: function(data) {
+            var values = {};
+            var notSet = []; // List of field names that should be set to default values
+            for (var i in fields) {
+                var field = fields[i];
+                if (field.isSet()) {
+                    values[field.name] = field.val();
+                } else {
+                    notSet.push(field.name);
+                }
+                // Remove TinyMCE instances to make sure jQuery does not try to access stale instances
+                // when loading editor for another block:
+                if (field.hasEditor()) {
+                    field.removeEditor();
+                }
+            }
+            // If WYSIWYG editor was used,
+            // prefer its data over values of "Vectors" and "Expected result" fields:
+            if (!_.isEmpty(data)) {
+                values.vectors = JSON.stringify(data.vectors, undefined, 4);
+                values.expected_result_positions = data.expected_result_positions;
+                values.expected_result = JSON.stringify(data.expected_result, undefined, 4);
+            }
+
+            studio_submit({values: values, defaults: notSet});
+        },
+
+        cancel: function() {
+            // Remove TinyMCE instances to make sure jQuery does not try to access stale instances
+            // when loading editor for another block:
+            for (var i in fields) {
```

Comment:
Why not use [`_.each()`](http://underscorejs.org/#each) or [`$.each()`](http://api.jquery.com/jQuery.each/) instead?